api/src/main/java/edu/cornell/mannlib/vitro/webapp/dao/filtering/filters/FilterByDisplayPermission.java

```diff
@@ -15,69 +17,56 @@
 import edu.cornell.mannlib.vitro.webapp.auth.identifier.IdentifierBundle;
 import edu.cornell.mannlib.vitro.webapp.auth.identifier.RequestIdentifiers;
 import edu.cornell.mannlib.vitro.webapp.auth.identifier.common.HasPermission;
-import edu.cornell.mannlib.vitro.webapp.auth.permissions.DisplayByRolePermission;
 import edu.cornell.mannlib.vitro.webapp.auth.permissions.Permission;
 import edu.cornell.mannlib.vitro.webapp.auth.permissions.PermissionRegistry;
 import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.RequestedAction;
 import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.display.DisplayDataProperty;
 import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.display.DisplayDataPropertyStatement;
 import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.display.DisplayObjectProperty;
 import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.display.DisplayObjectPropertyStatement;
-import edu.cornell.mannlib.vitro.webapp.beans.DataProperty;
-import edu.cornell.mannlib.vitro.webapp.beans.DataPropertyStatement;
-import edu.cornell.mannlib.vitro.webapp.beans.ObjectProperty;
-import edu.cornell.mannlib.vitro.webapp.beans.ObjectPropertyStatement;
 
 /**
  * Filter the properties depending on what DisplayByRolePermission is on the
  * request. If no request, or no permission, use the Public permission.
  */
-public class FilterByRoleLevelPermission extends VitroFiltersImpl {
-	private static final Log log = LogFactory
-			.getLog(FilterByRoleLevelPermission.class);
+public class FilterByDisplayPermission extends VitroFiltersImpl {
+	private static final Log log = LogFactory.getLog(FilterByDisplayPermission.class);
 
-	private final Permission permission;
+	private final String permissionSetUri;
 
-	private static Permission getDefaultPermission(ServletContext ctx) {
-		if (ctx == null) {
-			throw new NullPointerException("context may not be null.");
-		}
-
-		return PermissionRegistry.getRegistry(ctx).getPermission(
-				DisplayByRolePermission.NAMESPACE + "Public");
-	}
+	// Set a default permission set uri
+	private static final String defaultPermissionSetUri = "";
 
-	private static Permission getPermissionFromRequest(HttpServletRequest req) {
+	private static String getPermissionSetFromRequest(HttpServletRequest req) {
 		if (req == null) {
 			throw new NullPointerException("request may not be null.");
 		}
 
 		IdentifierBundle ids = RequestIdentifiers.getIdBundleForRequest(req);
-		for (Permission p : HasPermission.getPermissions(ids)) {
-			if (p instanceof DisplayByRolePermission) {
-				return p;
-			}
+		for (String uri : HasPermissionSet.getPermissionSetUris(ids)) {
+			return uri;
```

Comment:
Since this `for` loop is not looping, the logic may be more clear to simply return the first element if the permission-set is not empty.